sources/api/netdog/src/networkd/config/netdev.rs

```diff
@@ -111,3 +116,32 @@ impl Display for ArpAllTargets {
         }
     }
 }
+
+impl NetDevConfig {
+    const FILE_EXT: &str = "netdev";
+
+    /// Write the config to the proper directory with the proper prefix and file extention
+    pub(crate) fn write_config_file(&self) -> Result<()> {
+        let cfg_path = self.config_path()?;
+
+        fs::write(&cfg_path, self.to_string()).context(error::NetworkDConfigWriteSnafu {
+            what: "netdev_config",
+            path: cfg_path,
+        })
+    }
+
+    /// Build the proper prefixed path for the config file
+    fn config_path(&self) -> Result<PathBuf> {
+        let device_name = &self.netdev.as_ref().and_then(|n| n.name.clone()).context(
+            error::ConfigMissingNameSnafu {
+                what: "netdev config".to_string(),
+            },
+        )?;
+
+        let filename = format!("{}{}", CONFIG_FILE_PREFIX, device_name);
+        let mut path = Path::new(NETWORKD_CONFIG_DIR).join(filename);
```

Comment:
If you pass in `NETWORKD_CONFIG_DIR` then file writing and path generation becomes test-able.